packages/dx-react-scheduler/src/plugins/integrated-grouping.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import { Plugin, Getter } from '@devexpress/dx-react-core';
+import { memoize } from '@devexpress/dx-core';
+import {
+  getGroupingItemsFromResources, expandViewCellsDataWithGroups,
+  sortFilteredResources, filterResourcesByGrouping, updateGroupingWithMainResource,
+} from '@devexpress/dx-scheduler-core';
+import { IntegratedGroupingProps } from '../types';
+
+const pluginDependencies = [
+  { name: 'Resources' },
+  { name: 'GroupingState' },
+  { name: 'DayView', optional: true },
+  { name: 'MonthView', optional: true },
+  { name: 'WeekView', optional: true },
+];
+
+const getViewCellsDataComputed = memoize((
+  { viewCellsData, groupingItems, resourcesToGroupBy },
+) => expandViewCellsDataWithGroups(viewCellsData, groupingItems, resourcesToGroupBy));
+
+const getGroupingItemsComputed = memoize((
+  { resourcesToGroupBy },
+) => getGroupingItemsFromResources(resourcesToGroupBy));
+
+const getResourcesToGroupByComputed = memoize((
+  { resources, grouping },
+) => sortFilteredResources(filterResourcesByGrouping(resources, grouping), grouping));
+
+const getGroupingComputed = memoize((
+  { grouping, resources },
+) => updateGroupingWithMainResource(grouping, resources));
+
+const IntegratedGroupingBase: React.SFC<IntegratedGroupingProps> = React.memo(() => (
```

Comment:
Should we use `memo` here? This plugin doesn't use `props`

Reply:
@MaximKudriavtsev,

After drag this plugin rerenders if we don't use `memo`. We update data and call `Scheduler` plugin rerender.